docs/_documentation/02_features/built-in-functions.md

```diff
@@ -0,0 +1,437 @@
+      └── terragrunt.hcl
+
+Then `assets.txt` could be read with the following function call:
+
+``` hcl
+file("assets/mysql/assets.txt")
+```
+
+### find\_in\_parent\_folders
+
+`find_in_parent_folders()` searches up the directory tree from the current `terragrunt.hcl` file and returns the relative path to the first `terragrunt.hcl` in a parent folder or exit with an error if no such file is found. This is primarily useful in an `include` block to automatically find the path to a parent `terragrunt.hcl` file:
+
+``` hcl
+include {
+  path = find_in_parent_folders()
+}
+```
+
+The function takes an optional `name` parameter that allows you to specify a different filename to search for:
+
+``` hcl
+include {
+  path = find_in_parent_folders("some-other-file-name.hcl")
+}
+```
+
+You can also pass an optional second `fallback` parameter which causes the function to return the fallback value (instead of exiting with an error) if the file in the `name` parameter cannot be found:
+
+``` hcl
+include {
+  path = find_in_parent_folders("some-other-file-name.tfvars", "fallback.tfvars")
+}
+```
+
+### path\_relative\_to\_include
+
+`path_relative_to_include()` returns the relative path between the current `terragrunt.hcl` file and the `path` specified in its `include` block. For example, consider the following folder structure:
+
+    ├── terragrunt.hcl
+    └── prod
+        └── mysql
+            └── terragrunt.hcl
+    └── stage
+        └── mysql
+            └── terragrunt.hcl
+
+Imagine `prod/mysql/terragrunt.hcl` and `stage/mysql/terragrunt.hcl` include all settings from the root `terragrunt.hcl` file:
+
+``` hcl
+include {
+  path = find_in_parent_folders()
+}
+```
+
+The root `terragrunt.hcl` can use the `path_relative_to_include()` in its `remote_state` configuration to ensure each child stores its remote state at a different `key`:
+
+``` hcl
+remote_state {
+  backend = "s3"
+  config = {
+    bucket = "my-terraform-bucket"
+    region = "us-east-1"
+    key    = "${path_relative_to_include()}/terraform.tfstate"
+  }
+}
+```
+
+The resulting `key` will be `prod/mysql/terraform.tfstate` for the prod `mysql` module and `stage/mysql/terraform.tfstate` for the stage `mysql` module.
+
+### path\_relative\_from\_include
+
+`path_relative_from_include()` returns the relative path between the `path` specified in its `include` block and the current `terragrunt.hcl` file (it is the counterpart of `path_relative_to_include()`). For example, consider the following folder structure:
+
+    ├── sources
+    |  ├── mysql
+    |  |  └── *.tf
+    |  └── secrets
+    |     └── mysql
+    |         └── *.tf
+    └── terragrunt
+      └── common.tfvars
+      ├── mysql
+      |  └── terragrunt.hcl
+      ├── secrets
+      |  └── mysql
+      |     └── terragrunt.hcl
+      └── terragrunt.hcl
+
+Imagine `terragrunt/mysql/terragrunt.hcl` and `terragrunt/secrets/mysql/terragrunt.hcl` include all settings from the root `terragrunt.hcl` file:
```

Comment:
Sanity check: GitHub shows this all in Italics. Bug or rendering issue?

Reply:
Github makes it italics because of `*` (star) in: `└── *.tf`. I added an escape character `\` so it should fix the problem.